Subscriber/AbstractSubscriber.php

```diff
@@ -39,18 +50,19 @@ public function __construct(Container $container, $path = null)
 
     /**
      * Returns an array in the form of var association:
-     * list($controller, $request, $view) = self::getEverytihngFromArgs($args);
-     *
-     * @param $args
+     * list($controller, $request, $view) = self::getEverythingFromArgs($args);
      *
-     * @return array
+     * @param \Enlight_Event_EventArgs $args
+     * @param null $element
+     * @return array|mixed|null
      */
-    final public static function getEverytihngFromArgs ($args)
+    final public static function getEverythingFromArgs(\Enlight_Event_EventArgs $args, $element = null)
     {
         /**
          * @var \Enlight_Controller_Action $controller
          */
-        $controller = $args->getSubject();
+        $controller = $args->get('subject');
```

Comment:
Not necessary, since both calls work. My version uses the magic __call method on that class. If this is ok for you, please leave this line unchanged.

Reply:
IMHO that's like using Shopware()->Models() or $container->get('models')

But using magic method calls results in "yellow" marked/ highlighted code in my IDE and I don't like that :-P

Reply:
Ok, well since both solutions point to the same internally method call, I will not continue discussing on that. Accepted! ;-)